File: 1813-sentence-similarity-iii/1813-sentence-similarity-iii.js
/**
 * @param {string} sentence1
 * @param {string} sentence2
 * @return {boolean}
 */
var areSentencesSimilar = function(sentence1, sentence2) {
  const s1 = sentence1.split(" ");
  const s2 = sentence2.split(" ");
  
  while (s1.length && s2.length && s1[0] == s2[0]) {
    s1.shift();
    s2.shift();
  }
  
  while (s1.length && s2.length && s1[s1.length - 1] == s2[s2.length - 1]) {
    s1.pop();
    s2.pop();
  }
  
  return !s1.length || !s2.length;
};

File: 1813-sentence-similarity-iii/1813-sentence-similarity-iii.py
class Solution(object):
    def areSentencesSimilar(self, sentence1, sentence2):
        """
        :type sentence1: str
        :type sentence2: str
        :rtype: bool
        """
        s1 = list(sentence1.split())
        s2 = list(sentence2.split())
        
        while s1 and s2 and s1[0] == s2[0]:
            s1.pop(0)
            s2.pop(0)
          
        while s1 and s2 and s1[-1] == s2[-1]:
            s1.pop(-1)
            s2.pop(-1)
          
        return not s1 or not s2
        